### PR TITLE
Dashboard line

### LINKED ISSUE #53: Dashboard - include line graphs
- [x] Create selectbox for filtering graph
- [x] Query data needed for line graph using filter
- [x] Write function to create graph
- [x] Call function to create graph on app

As an engineer, to adhere to stakeholder requirements, I should create the necessary visualisations to best represent the data

### PR DESCRIPTION
**Pull request by alina-101**
Added line graphs for sales per album and sales per tag.
Fixed duplicate album names showing up in album titles list.
closes #53

### CODE REVIEW

File: dashboard/database.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def get_sales_by_country(_conn: Connection, n: int = 5):
     return data
 
 
-def get_all_album_titles(_conn: Connection):
+def get_all_album_purchase_titles(_conn: Connection):
```
Comment (e-lemma) [inline dashboard/database.py:183]: Missing an expected return hint

```suggestion
def get_all_album_purchase_titles(_conn: Connection) -> pd.DataFrame:
```

I think thats the correct one?

File: dashboard/database.py [hunk 2]
```diff
@@ -218,4 +218,54 @@ def get_album_sales_by_album(_conn: Connection, album_name: str):
         cur.execute(query, (album_name, ))
         data = cur.fetchall()
 
-    return data
+    return pd.DataFrame(data)
+
+
+def get_all_tag_names(_conn: Connection):
```
Comment (e-lemma) [inline dashboard/database.py:224]: Missing a return hint

```suggestion
def get_all_tag_names(_conn: Connection) -> list[str]:
```

File: dashboard/database.py [hunk 3]
```diff
@@ -218,4 +218,54 @@ def get_album_sales_by_album(_conn: Connection, album_name: str):
...
+        """
+
+    with _conn.cursor() as cur:
+        cur.execute(query)
+        data = cur.fetchall()
+
+    return [d["name"] for d in data]
+
+
+def get_tag_sales_by_tag(_conn: Connection, tag_name: str):
```
Comment (e-lemma) [inline dashboard/database.py:242]:
```suggestion
def get_tag_sales_by_tag(_conn: Connection, tag_name: str) -> pd.DataFrame:
```

File: dashboard/database.py
Comment (Zhi-704) [inline dashboard/database.py:1]: I believe your query for popular artist isn't correct. You're grouping it by album and you're also not counting distinct purchases, meaning duplicate data is also considered.

Comment (alina-101) [reply dashboard/database.py:1]: That's true, I didn't spot that. Issue #62 addresses this.

**Review by e-lemma — APPROVED** [review]
Comments have been addressed and everything looks good, just make sure to fix #61 in another PR

**Pull request merged**